src/Metalama.Patterns.NotifyPropertyChanged/Implementation/Natural/BuildAspectContext.cs

```diff
@@ -0,0 +1,451 @@
+    }
+
+    // ReSharper disable once UnusedAutoPropertyAccessor.Global
+    public bool InsertDiagnosticComments { get; set; } // = true; // TODO: Set by configuration? Discuss.
+
+    public IAspectBuilder<INamedType> Builder { get; }
+
+    public INamedType Target => this.Builder.Target;
+
+    public bool TargetImplementsInpc { get; }
+
+    // ReSharper disable once MemberCanBePrivate.Global
+    public bool BaseImplementsInpc { get; }
+
+    public IMethod? BaseOnPropertyChangedMethod => this._baseOnPropertyChangedMethod.Value;
+
+    public IMethod? BaseOnChildPropertyChangedMethod => this._baseOnChildPropertyChangedMethod.Value;
+
+    public IMethod? BaseOnUnmonitoredObservablePropertyChangedMethod => this._baseOnUnmonitoredObservablePropertyChangedMethod.Value;
+
+    public bool HasInheritedOnChildPropertyChangedPropertyPath( string parentPropertyPath )
+    {
+        this._inheritedOnChildPropertyChangedPropertyPaths ??=
+            BuildPropertyPathLookup( GetPropertyPaths( this.Elements.OnChildPropertyChangedMethodAttribute, this.BaseOnChildPropertyChangedMethod ) );
+
+        return this._inheritedOnChildPropertyChangedPropertyPaths.Contains( parentPropertyPath );
+    }
+
+    public bool HasInheritedOnUnmonitoredObservablePropertyChangedProperty( string propertyName )
+    {
+        this._inheritedOnUnmonitoredObservablePropertyChangedPropertyNames ??=
+            BuildPropertyPathLookup(
+                GetPropertyPaths( this.Elements.OnUnmonitoredObservablePropertyChangedMethodAttribute, this.BaseOnUnmonitoredObservablePropertyChangedMethod ) );
+
+        return this._inheritedOnUnmonitoredObservablePropertyChangedPropertyNames.Contains( propertyName );
+    }
+
+    public CertainDeferredDeclaration<IMethod> OnPropertyChangedMethod { get; } = new();
+
+    public CertainDeferredDeclaration<IMethod> OnChildPropertyChangedMethod { get; } = new();
+
+    public DeferredDeclaration<IMethod> OnUnmonitoredObservablePropertyChangedMethod { get; } =
+        new( willBeDefined: true ); // TODO: Decide according to configuration.
+
+    public List<string> PropertyPathsForOnChildPropertyChangedMethodAttribute { get; } = new();
+
+    public List<string> PropertyNamesForOnUnmonitoredObservablePropertyChangedMethodAttribute { get; } = new();
+
+    /// <summary>
+    /// Gets the <see cref="IProperty"/> for property <c>EqualityComparer<paramref name="type"/>>.Default</c>.
+    /// </summary>
+    /// <param name="type"></param>
+    /// <returns></returns>
+    public IProperty GetDefaultEqualityComparerForType( IType type )
+        => this.Elements.EqualityComparerOfT.WithTypeArguments( type ).Properties.Single( p => p.Name == "Default" );
+
+    private DependencyGraphNode? _dependencyGraph;
+    private bool _prepareDependencyGraphReportedErrors;
+
+    public DependencyGraphNode PrepareDependencyGraph()
+    {
```

Comment:
The dependency graph should normally not be computed at design time because it is expensive and mostly useless to support the design-time experience. It could be computed for the "preview" and "live template" scenario but it should not be called by default at design time.